core/src/main/java/hudson/model/Queue.java

```diff
@@ -1442,7 +1442,10 @@ public void maintain() {
                     if (r != null) {
                         r.run();
                     } else {
-                        new BlockedItem(top).enter(this);
+                        //this is to solve JENKINS-30084: the task has to be buildable to force
```

Comment:
I *think* this is incorrect.
The job could be blocked for reasons other than no nodes with the correct label are available.

(concurrent builds is disabled for example - or many other reasons).